modules/internal/Bytes.chpl

```diff
@@ -0,0 +1,473 @@
+
+/*
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+The following documentation shows functions and methods used to
+manipulate and process Chapel bytes objects.
+
+Casts from bytes to a Numeric Type
+-----------------------------------
+
+This module supports casts from :record:`bytes` to numeric types. Such casts
+will convert the bytes to the numeric type and throw an error if the bytes
+is invalid. For example:
+
+.. code-block:: chapel
+
+  var number = "a":int;
+
+throws an error when it is executed, but
+
+.. code-block:: chapel
+
+  var number = "1":int;
+
+stores the value ``1`` in ``number``.
+
+To learn more about handling these errors, see the
+:ref:`Error Handling technical note <readme-errorHandling>`.
+ */
+module Bytes {
+  use ChapelStandard;
+
+  // to follow String.chpl, and maybe limit the index to smaller integers
+  type byteIndex = int; 
+
+  type byteType = uint(8);
+  type bufferType = c_ptr(byteType);
```

Comment:
Not relevant for API review, but `string` has a home locale and `bytes` could do that too. Or, even better, we could actually go and implement `wide_c_ptr` / `c_wide_ptr` (#12490) and use that in both places.